Fix and improve crud test

mkahvi opened the pull request:
Implements actor type selection ~~and synthetic actor toggle~~. Also small improvements to related code thrown in.

Fixes #39

akrigline commented on module/classes/DevModePerformance.mjs:
```diff
@@ -17,14 +17,18 @@ export class DevModePerformance {
 
-  static actorCRUDTest = async (iterations = 1000) => {
-    console.log('running test with iterations:', `${iterations}`, !!iterations, iterations || 1000);
+  static actorCRUDTest = async (type, iterations = 1000, synthetic) => {
```
Since this is getting more complex, lets destructure an object as the first parameter here. That will help this 'self-document' as well.

```suggestion
  static actorCRUDTest = async ({ type, iterations = 1000, synthetic } = {}) => {
```

akrigline commented on module/classes/DevModeConfig.mjs:
```diff
@@ -188,9 +192,12 @@ export class DevModeConfig extends FormApplication {
+            const iterations = form.actorCrudIterations;
+            const synthetic = form.actorCrudSynthetic;
+            console.log('CRUD', { actorType, iterations, synthetic });
+            return DevModePerformance.actorCRUDTest(actorType, iterations || undefined, synthetic);
```
If we use an object as the param for `actorCRUDTest`, we might be able to pass the output of `formData` directly into this method.

mkahvi replied:
Giving form data directly would give really ugly parameter names, unless the CRUD test is given its own form. That gets into rather large redesigns of the settings dialog.

mkahvi replied:
I think I had an idea for a solution.

mkahvi commented:
I removed synthetic option as I discovered it wasn't used correctly, and after I fixed it, I also realized Foundry explodes on .update() calls to such. So even if the idea is great (IMO), Foundry doesn't support it.

akrigline commented on module/classes/DevModePerformance.mjs:
```diff
@@ -17,14 +17,17 @@ export class DevModePerformance {
 
-  static actorCRUDTest = async (iterations = 1000) => {
-    console.log('running test with iterations:', `${iterations}`, !!iterations, iterations || 1000);
+  static actorCRUDTest = async (type, iterations = 1000) => {
```
```suggestion
  static actorCRUDTest = async ({ type, iterations = 1000} = {}) => {
```
by feeding it the object output from `FormDataExtended`, this will need to destructure that object

mkahvi replied:
I thought I did that already... I honestly had problems with git that might've lost that.

akrigline commented on module/classes/DevModePerformance.mjs:
```diff
@@ -17,14 +17,17 @@ export class DevModePerformance {
+  static actorCRUDTest = async (type, iterations = 1000) => {
+    if (!game.system.template.Actor.types.includes(type)) return console.error(type, "is invalid actor type");
+    // Force some defaults
+    iterations ||= 1000;
```
```suggestion
```

This is handled in the params and shouldn't be necessary here.

akrigline reviewed (changes requested):
Shame that the temporary thing didn't work out. Still thanks for setting up the better system support
